src/provision.js

```
@@ -0,0 +1,148 @@
+const csv = require('csv-parser');
+const fs = require('fs');
+
+const platformClient = require('purecloud-platform-client-v2');
+const groupsApiProxy = require('./proxies/groupsapi');
+const usersApiProxy = require('./proxies/usersapi');
+const sitesApiProxy = require('./proxies/sitesapi');
+const phoneBaseApiProxy = require('./proxies/phonebaseapi');
+const phoneApiProxy = require('./proxies/phoneapi');
+const rolesApiProxy = require('./proxies/rolesapi');
+const stationsApiProxy = require('./proxies/stationsapi');
+
+/*Takes a list of users (sourced from a csv file) and assigns them to a chat group*/
+const assignUsersToGroups = async (users) => {
+  const groupIds = groupsApiProxy.getGroupIds();
+
+  groupIds.forEach(async (groupId) => {
+    //Find each user in a group
+    const userIdsInGroup = users
+      .filter((user) => groupId === user.groupId)
+      .map((user) => user.userId);
+
+    if (userIdsInGroup.length > 0) {
+      try {
+        await groupsApiProxy.addUsersToAGroup(groupId, userIdsInGroup);
+      } catch (e) {
+        console.log(
```

Should this stop execution or be handled elsewhere with more than a log message? If so you should probably rethrow the error.

Also its better to use console.error instead of console.log for error messages.

---

Yeah.  I purposely call out in the tutorial that this is not the best error handling.  Interested to hear your thoughts.  Also will fix with console.error